plugins/inputs/slurm/slurm.go

```diff
@@ -0,0 +1,508 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package slurm
+
+import (
+	"context"
+	stdTls "crypto/tls"
+	_ "embed"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+
+	goslurm "github.com/pcolladosoto/goslurm/v0038"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type Slurm struct {
+	URL              string          `toml:"url"`
+	Username         string          `toml:"username"`
+	Token            string          `toml:"token"`
+	IgnoredEndpoints []string        `toml:"ignored_endpoints"`
+	ResponseTimeout  config.Duration `toml:"response_timeout"`
+	tls.ClientConfig
+
+	client      *goslurm.APIClient
+	baseURL     *url.URL
+	endpointMap map[string]bool
+}
+
+func (*Slurm) SampleConfig() string {
+	return sampleConfig
+}
+
+func (s *Slurm) Init() error {
+	if s.ResponseTimeout == 0 {
+		s.ResponseTimeout = config.Duration(time.Second * 5)
+	}
+
+	s.endpointMap = map[string]bool{}
+	for _, endpoint := range s.IgnoredEndpoints {
+		s.endpointMap[strings.ToLower(endpoint)] = true
+	}
+
+	if s.URL == "" {
+		return errors.New("empty URL provided")
+	}
+
+	u, err := url.Parse(s.URL)
+	if err != nil {
+		return err
+	}
+
+	if u.Hostname() == "" {
+		return fmt.Errorf("empty hostname for url %q", s.URL)
+	}
+
+	s.baseURL = u
+
+	if u.Scheme != "http" && u.Scheme != "https" {
+		return fmt.Errorf("invalid scheme %q", u.Scheme)
+	}
+
+	var tlsCfg *stdTls.Config
+	if u.Scheme == "https" {
+		tlsCfg, err = s.ClientConfig.TLSConfig()
+		if err != nil {
+			return err
+		}
+	}
+
+	configuration := goslurm.NewConfiguration()
+	configuration.Host = u.Host
+	configuration.Scheme = u.Scheme
+	configuration.UserAgent = internal.ProductToken()
+	configuration.HTTPClient = &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsCfg,
+		},
+		Timeout: time.Duration(s.ResponseTimeout),
+	}
+
+	s.client = goslurm.NewAPIClient(configuration)
+
+	return nil
+}
+
+func (s *Slurm) parseTres(tres string) map[string]interface{} {
+	parsedValues := map[string]interface{}{}
+	var ok bool
```

Comment:
Please move this declaration into the `for` loop to limit its scope!

Reply:
Whops! Done on 47393078166c438b5a0f2263661b1170b5c02f67.